### PR TITLE
Added Collections/Wishlist functionality

### PR DESCRIPTION
**Pull request by echaulk2**

### CODE REVIEW

File: functions/collection.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    async createCollection(): Promise<Collection> {
+        let params = {
+            TableName: table,
+            Item: {
+              userID: this.userID,
+              sortKey: this.sortKey,
+              collectionType: this.collectionType,
+              games: this.games
+            },
+            ConditionExpression: 'attribute_not_exists(collectionType) AND attribute_not_exists(userID) AND attribute_not_exists(sortKey)'
```
[the remark this answers is not in the document]
Comment (leavertj) [reply functions/collection.ts:40]: Also looks like there is some cross-over here with Dynamo operations. I'd try to keep the persistence logic to one class if possible.

File: functions/collectionManager.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+export async function getCollection(userID: string, collectionType: string) : Promise<Collection> {
+    let sortKey = `[${userID}]#[collection]#[${collectionType}]`;
+    let params = {
+      TableName: table,
+      Key: {
+        userID: userID,
+        sortKey: sortKey
+      },
+      KeyConditionExpression: `sortKey = ${sortKey} AND userID = ${userID}`,
+      ConditionExpression: 'attribute_exists(sortKey) and attribute_exists(userID)'
```
Comment (leavertj) [inline functions/collectionManager.ts:26]: Not sure you need this condition

File: functions/collectionManager.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      //Not sure how to handle this since the documentClient doesn't throw an error
+      if (err.message == "Cannot read property 'userID' of undefined") {
+        throw new CollectionError("Unable to find collection", 404);
+      }
+      throw err
+    }
+}
+
+export async function addGameToCollection(game: Game, collection: Collection) : Promise<Collection> {
+    collection.addGame(game);
```
Comment (leavertj) [inline functions/collectionManager.ts:44]: what happens if a user tries to add two games in quick succession? Quick enough that two instances of 'addGameToCollection' were executing at the same time attempting to add two different games.

File: functions/collection.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+const table = (isTest) ? process.env.DYNAMO_DB_TEST_TABLE : process.env.DYNAMO_DB_GAME_TABLE;
+import { Game } from "./game";
+import { IDynamoObject } from "./gameManager";
+import { getCollection } from "./collectionManager";
+import { CollectionError } from "./collectionErrorHandler";
+
+export class Collection {
+    userID: string;
+    sortKey: string;    
+    collectionType: string;
```
Comment (leavertj) [inline functions/collection.ts:21]: How would you use this attribute to extend the functionality of a wishlist? What if you want to add a few functions specifically for wishlists?

**Review by leavertj — CHANGES_REQUESTED** [review]

**Comment by echaulk2** [comment]
Implementing the changes in this PR in another PR

**Pull request closed without merging**